extmod/nimble/modbluetooth_nimble.c

```diff
@@ -846,7 +846,7 @@ STATIC int ble_gatt_attr_read_cb(uint16_t conn_handle, const struct ble_gatt_err
     if (error->status == 0) {
         gattc_on_data_available(MP_BLUETOOTH_IRQ_GATTC_READ_RESULT, conn_handle, attr->handle, attr->om);
     }
-    mp_bluetooth_gattc_on_read_write_status(MP_BLUETOOTH_IRQ_GATTC_READ_DONE, conn_handle, attr->handle, error->status);
+    mp_bluetooth_gattc_on_read_write_status(MP_BLUETOOTH_IRQ_GATTC_READ_DONE, conn_handle, attr ? attr->handle : -1, error->status);
```

Comment:
Thanks @friebel !

Does the same change need to be made in `ble_gatt_attr_write_cb`?

Reply:
Hello Jim,

> Does the same change need to be made in `ble_gatt_attr_write_cb`?

No, the write handler is okay. NimBLE always calls the write handler with a non-NULL `attr`, see [nimble/host/src/ble_gattc.c#L3446](https://github.com/apache/mynewt-nimble/blob/6e7c1d5d97ef02fda9919d4e83d407ac64cd26f9/nimble/host/src/ble_gattc.c#L3446).